### PR TITLE
[PSCmdAssistant] Grant-AzDiskAccess new parameter SecureVMGuestStateSAS

### PR DESCRIPTION
**Pull request by Sandido**
## Description
This one is a typical commit flow.
Feature implementation of: https://msazure.visualstudio.com/One/_workitems/edit/27063119
Design request: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1409

Added the new parameter SecureVmGuestStateSas to Grant-AzDiskAccess to return the securityDataAccessSAS value.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by Sandido** [comment]
@Nickcandy , the test failure in ssh seems due to an error when trying to install from a repo PsGallery and failed to do so. I don't see how my changes would affect that.

**Comment by Nickcandy** [comment]
Hi @Sandido, when your pr is ready to review, please set the pr "ready to review"

**Comment by Sandido** [comment]
@Nickcandy , just reviewing it with my team first.

File: src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1 [hunk 1]
```diff
@@ -1825,4 +1825,46 @@ function Test-SnapshotConfigTierOptionEnhancedSpeed
...
+Using a TL disk, use the SecureVmGuestStateSas parameter to get the securityDataAccessSAS value.
+#>
+function Test-DiskGrantAccessGetSASWithTL
+{
+    $rgname = Get-ComputeTestResourceName;
+	$loc = Get-ComputeVMLocation;#'eastus';
+
+	try
+    {
+        # $rgname = "adsandtlg4";
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1:1841]: cleanup

File: src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1 [hunk 2]
```diff
@@ -1825,4 +1825,46 @@ function Test-SnapshotConfigTierOptionEnhancedSpeed
...
+        # $loc = "eastus";
+		New-AzResourceGroup -Name $rgname -Location $loc -Force;
+
+        $diskname = "d" + $rgname;
+        $securityTypeTL = "TrustedLaunch";
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1:1846]: not needed. Could consider a test for a nonTL disk, but not specific to PS functionality more a test that the backend is returning what we expect.

File: src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1 [hunk 3]
```diff
@@ -1825,4 +1825,46 @@ function Test-SnapshotConfigTierOptionEnhancedSpeed
...
+        $disk = New-AzDisk -ResourceGroupName $rgname -DiskName $diskname -Disk $diskconfig;
+
+        $grantAccess = Grant-AzDiskAccess -ResourceGroupName $rgname -DiskName $diskname -Access 'Read' -DurationInSecond 60 -SecureVMGuestStateSAS;
+        Assert-NotNull $grantAccess.securityDataAccessSAS;
+        Assert-NotNull $grantAccess.AccessSAS;
+
+        $grantAccess = Grant-AzDiskAccess -ResourceGroupName $rgname -DiskName $diskname -Access 'Read' -DurationInSecond 60;
+        Assert-NotNull $grantAccess.AccessSAS;
+        Assert-Null $grantAccess.securityDataAccessSAS;
+
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1:1863]: create a vm here just in case

File: src/Compute/Compute/ChangeLog.md [hunk 4]
```diff
@@ -20,6 +20,7 @@
 
 -->
 ## Upcoming Release
+* Added new optional parameter `SecureVmGuestStateSas` to cmdlet `Grant-AzDiskAccess`.
```
Comment (Sandido) [inline src/Compute/Compute/ChangeLog.md:23]: capitalization correction

File: src/Compute/Compute/Generated/Models/PSAccessUri.cs [hunk 5]
```diff
@@ -29,5 +29,7 @@ public partial class PSAccessUri
     {
         public string AccessSAS { get; set; }
 
+        public string SecurityDataAccessSAS { get; private set; }
```
Comment (Sandido) [inline src/Compute/Compute/Generated/Models/PSAccessUri.cs:32]: AccessSAS also does not have private despite having it in SDK. Keep it consistent for now, let users modify PS objects, SDK won't let them modify the others.

**Review by haagha — APPROVED** [review]
@Sandido to add vm creation to Test and change "private set" to "set" but other than that everything looks good!

**Comment by Sandido** [comment]
Ready for merging

**Comment by Nickcandy** [comment]
Hi @Sandido  , our code freeze time is PST 6PM Apr 22. If your code targets to this release, please finish them ASAP

**Comment by Sandido** [comment]
@Nickcandy , resolved. Looks like something tried to add back in the bug of ProgressAction paramter to the help doc.

**Comment by Sandido** [comment]
@Nickcandy the error in the 'azure-powershell security tools' check is no due to anything related to my edits.

**Comment by Nickcandy** [comment]
image
seems you delete the end symbol leading to build error

**Comment by YanaXu** [comment]
@sandido, could you please resolve the conflicts?

**Comment by Sandido** [comment]
@YanaXu , resolved. Please merge when you can. This constant issue of conflicts with Changelog.md is quite annoying.

**Review by YanaXu — APPROVED** [review]

**Pull request merged**